src/dict_sorting.jl

```diff
@@ -19,3 +19,16 @@ end
 
 sort(d::OrderedDict; args...) = sort!(copy(d); args...)
 sort(d::Dict; args...) = sort!(OrderedDict(d); args...)
+
+function sort!(d::LittleDict; byvalue::Bool=false, args...)
+    if byvalue
+        p = sortperm(d.vals; args...)
+    else
+        p = sortperm(d.keys; args...)
+    end
+    d.keys = d.keys[p]
```

Comment:
`d.keys .= d.keys[p]` to save a copy? Of course that won't work for tuples, probably need to dispatch on `FrozenLittleDict` (which would be a typealias).

Reply:
I don't see the advantage.

We have two vectors,  `d.keys[p]` and `d.keys`.

If we have `d.keys = d.keys[p]`, then the old `d.keys` will be freed and can be garbage collected, and `d.keys[p]` will be maintained.

If we have `d.keys .= d.keys[p]`, then `d.keys[p]` will go out of scope at end of functioon  and can be garbage collected, and `d.keys` will continue to be maintained.
Also we will have done an extra copy of each element to move it from one to the other.

Either way we are allocating and garbage collecting the same amount of memory.

Reply:
`LittleDict` is not mutable so it's not possible to replace its fields. Why are you not getting `ERROR: setfield! immutable struct of type LittleDict cannot be changed`?

Reply:
Good point. I'm halfway though porting over the tests. I though `sort` was passing.
but  it clearly can't have been

Reply:
I have not defined `sort!` and just have done `sort`.